test/ion_assert.c

```diff
@@ -0,0 +1,5 @@
+#include "ion_assert.h"
+
+iERR assert_equals_int(int expected, int actual) {
+    return (expected == actual) ? IERR_OK : IERR_INVALID_ARG;
+}
```

Comment:
I'd just fold this into the header and make it `static inline` no sense for an extra file and translation unit for something so trivial.